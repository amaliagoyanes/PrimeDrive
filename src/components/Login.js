import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Platform,
    Text, TextInput,
    View, AsyncStorage,
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import DeviceInfo from 'react-native-device-info';
import {Col, Row, Grid} from 'react-native-easy-grid';
import Immersive from 'react-native-immersive';
import {Actions} from 'react-native-router-flux';
import {createIconSetFromIcoMoon} from 'react-native-vector-icons';
import icoMoonConfig from '../fonts/selection.json';
const Icon = createIconSetFromIcoMoon(icoMoonConfig);

import {guid} from './index'
import styles from '../style/styles';
import firestack from '../fbconfig.js'
import {scale, scaleText} from '../scaling';

export default class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            error: false, errorMessage: ''
        };

        this.login = this.login.bind(this);
    }

    componentDidMount() {
        if (Platform.OS !== 'ios') {
            Immersive.on();
            Immersive.setImmersive(true);
        }
    }

    login() {
        if (!this.state.email.length) {
            this._email.focus()
        } else if (!this.state.password.length) {
            this._pwd.focus()
        } else {
            firestack.auth.signInWithEmail(this.state.email, this.state.password)
                .then((user) => {
                    user.user.password = this.state.password;
                    let uuid = DeviceInfo.getUniqueID();
                    let name = DeviceInfo.getDeviceName();
                    if (name === 'Unknown')
                        name = DeviceInfo.getModel();
                    if (!name || name === undefined) {
                        name = 'Tablet';
                    }
                    let data = {name: name,};
                    let token = Platform.OS !== 'ios' ? user.user.token : user.user.refreshToken;
                    firestack.database.ref(`/clients/${uuid}`).set(data).then(response => {
                        try {
                            AsyncStorage.setItem('@PrimeDrive:user', JSON.stringify(user)).then(() => {
                                AsyncStorage.setItem('@PrimeDrive:client', uuid).then(() => {
                                    AsyncStorage.setItem('@PrimeDrive:token', JSON.stringify(token))
                                        .then(() => Actions.home({from: 'login'})).catch();
                                }).catch();
                            });
                        } catch (error) {
                            // Error saving data
                            console.log('error saving', error)
                        }
                    }).catch(err => console.log(err))
                })
                .catch((err) => {
                    console.log('User signin error', err);
                    this.setState({error: true});
                })
        }
    }

    render() {
        return (
            <Grid style={[styles.container, {paddingLeft: 30, paddingTop: scale(25)}]}>
                <Row/>
                <Row size={0.75}>
                    <Col style={{alignItems: 'center'}}>
                        <Icon style={[styles.icon, {top: 0}]} name="logo" size={scaleText(72)} color="white"/>
                    </Col>
                </Row>
                <Row size={0.25}>
                    {this.state.error ?
                        <Text style={[styles.text, {color: '#e84b3a'}]}>
                            <FormattedMessage id="app.incorrectCredentials" defaultMessage="Incorrect username or password" />
                        </Text> :
                        <Text/>
                    }
                </Row>
                <Row size={0.5}>
                    <Col style={{alignItems: 'center'}}>
                        <TextInput
                            ref={elt => this._email = elt}
                            style={styles.loginInput} keyboardType="email-address"
                            underlineColorAndroid={this.state.error ? '#e84b3a' : '#2dab61'}
                            onChangeText={(text) => this.setState({email: text, error: false})}
                            placeholder={'Email'} placeholderTextColor='#cecece' value={this.state.email}
                        />
                    </Col>
                </Row>
                <Row size={0.5}>
                    <Col style={{alignItems: 'center'}}>
                        <TextInput
                            ref={elt => this._pwd = elt}
                            style={styles.loginInput} secureTextEntry={true}
                            underlineColorAndroid={this.state.error ? '#e84b3a' : '#2dab61'}
                            onChangeText={(text) => this.setState({password: text, error: false})}
                            placeholder={'Password'} placeholderTextColor='#cecece' value={this.state.password}
                        />
                    </Col>
                </Row>
                <Row>
                    <Col style={{alignItems: 'center'}}>
                        <TouchableOpacity onPress={this.login}
                                          style={[styles.tabButton, styles.confirmBtn,]}>
                            <Text style={styles.text}><FormattedMessage id="app.login" defaultMessage="LOGIN" /></Text>
                        </TouchableOpacity>
                    </Col>
                </Row>
                <Row/>
            </Grid>
        )
    }
}
