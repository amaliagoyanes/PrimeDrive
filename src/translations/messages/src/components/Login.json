[
  {
    "id": "app.incorrectCredentials",
    "defaultMessage": "Incorrect username or password"
  },
  {
    "id": "app.login",
    "defaultMessage": "LOGIN"
  }
]