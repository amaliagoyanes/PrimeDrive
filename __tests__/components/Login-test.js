'use strict';

import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import Login from '../../src/components/Login';


Login.prototype.componentDidMount = jest.fn;
it('renders correctly', () => {
    const tree = shallow(
        <Login />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('TextInput').length).toEqual(2);
    expect(tree.find('TouchableOpacity').length).toEqual(1);
});

it('renders correctly with error', () => {
    const tree = shallow(
        <Login />);
    tree.setState({error: true});
    expect(tree).toMatchSnapshot();
});