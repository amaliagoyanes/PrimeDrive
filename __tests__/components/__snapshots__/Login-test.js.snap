// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<GridNB
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#3f3f48",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "paddingLeft": 30,
        "paddingTop": 25,
      },
    ]
  }
>
  <RowNB />
  <RowNB
    size={0.75}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="logo"
        size={72}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 5,
            },
            Object {
              "top": 0,
            },
          ]
        }
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={0.25}
  >
    <ColumnNB />
  </RowNB>
  <RowNB
    size={0.5}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#cecece"
        style={
          Object {
            "color": "#fff",
            "height": 40,
            "width": "50%",
          }
        }
        underlineColorAndroid="#2dab61"
        value=""
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={0.5}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#cecece"
        secureTextEntry={true}
        style={
          Object {
            "color": "#fff",
            "height": 40,
            "width": "50%",
          }
        }
        underlineColorAndroid="#2dab61"
        value=""
      />
    </ColumnNB>
  </RowNB>
  <RowNB>
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#404048",
              "justifyContent": "center",
              "width": "34%",
            },
            Object {
              "backgroundColor": "#2dab61",
              "height": 45,
              "width": "50%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Gotham-Book",
              "fontSize": 14,
            }
          }
        >
          LOGIN
        </Text>
      </TouchableOpacity>
    </ColumnNB>
  </RowNB>
  <RowNB />
</GridNB>
`;

exports[`renders correctly with error 1`] = `
<GridNB
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#3f3f48",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "paddingLeft": 30,
        "paddingTop": 25,
      },
    ]
  }
>
  <RowNB />
  <RowNB
    size={0.75}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="white"
        name="logo"
        size={72}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 5,
            },
            Object {
              "top": 0,
            },
          ]
        }
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={0.25}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "color": "#e84b3a",
          },
        ]
      }
    >
      Incorrect username or password
    </Text>
  </RowNB>
  <RowNB
    size={0.5}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#cecece"
        style={
          Object {
            "color": "#fff",
            "height": 40,
            "width": "50%",
          }
        }
        underlineColorAndroid="#e84b3a"
        value=""
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={0.5}
  >
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#cecece"
        secureTextEntry={true}
        style={
          Object {
            "color": "#fff",
            "height": 40,
            "width": "50%",
          }
        }
        underlineColorAndroid="#e84b3a"
        value=""
      />
    </ColumnNB>
  </RowNB>
  <RowNB>
    <ColumnNB
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#404048",
              "justifyContent": "center",
              "width": "34%",
            },
            Object {
              "backgroundColor": "#2dab61",
              "height": 45,
              "width": "50%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Gotham-Book",
              "fontSize": 14,
            }
          }
        >
          LOGIN
        </Text>
      </TouchableOpacity>
    </ColumnNB>
  </RowNB>
  <RowNB />
</GridNB>
`;
